API.rst

```diff
@@ -0,0 +1,161 @@
+The Skulpt Web Application API
+==============================
+
+This is a proposal for a new API to Skulpt, if adopted and implemented it will replace the current hodgepodge of functions currently used to embed Skulpt in an HTML web page.
+
+A few key preliminary points
+----------------------------
+
+* A smallish but more nicely and consistently named set of functions could wrap the current stuff and make life easier for people trying to get started.
+* The old functions would remain so as to provide backward compatibility, in fact I anticipate that this new api would largely be a thin wrapper over the old.  So keeping the old is important for backward compatibiliy and for advanced users who may want more fine grained control.
+
+Objects and Methods
+-------------------
+
+* skulpt.exec(code) -- handles promises, exceptions, calls skulpt.error when an error occurs...
+* skulpt.eval(code)  -- returns a javascript object resulting from evaling a Python expression
+* skulpt.input  --  function -- called when ``input`` is called a default with a dialog provided
+* skulpt.import -- function called for importing files the current builtinRead
+* skulpt.output = function   -- A default would be provided that would append text to a configured html element
+* skulpt.error = function  -- A default would be provided that would append text to a configured html element
```

Comment:
Excellent point.  I think I like the onInput idea, but I could be convinced either way.